p2pclient/streams.go

```diff
@@ -0,0 +1,109 @@
+package p2pclient
+
+import (
+	"errors"
+	"io"
+	"net"
+
+	ggio "github.com/gogo/protobuf/io"
+	pb "github.com/libp2p/go-libp2p-daemon/pb"
+)
+
+// NewStream initializes a new stream on one of the protocols in protos with
+// the specified peer.
+func (c *Client) NewStream(peer []byte, protos []string) (*pb.StreamInfo, io.ReadWriteCloser, error) {
```

Comment:
we might want to use `peer.ID` for the peer id instead of bytes?

Reply:
we might also want to tame the return type to a more friendly data structure that has parsed the response components.